### PR TITLE
Slightly better code around packets for parallel replicas

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Made some changes around packets which are used for communication for parallel reading from replicas (currently experimental feature).

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeReadPool.h [hunk 1]
```diff
@@ -254,8 +255,9 @@ class MergeTreeInOrderReadPoolParallelReplicas : private boost::noncopyable
             buffered_tasks.push_back({part.data_part->info, MarkRanges{}});
 
         extension.all_callback({
-            .description = parts_ranges.getDescriptions(),
-            .replica_num = extension.number_of_current_replica
+            mode,
```
Comment (nikitamikhaylov) [inline src/Storages/MergeTree/MergeTreeReadPool.h:258]: Fix is here...

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 3fab50ff12bcafc398af2c1d1e1c79851bf4cd41 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51451/3fab50ff12bcafc398af2c1d1e1c79851bf4cd41/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Storages/MergeTree/MergeTreeReadPool.h [hunk 2]
```diff
@@ -254,8 +255,9 @@ class MergeTreeInOrderReadPoolParallelReplicas : private boost::noncopyable
             buffered_tasks.push_back({part.data_part->info, MarkRanges{}});
 
         extension.all_callback({
```
Comment (antonio2368) [inline src/Storages/MergeTree/MergeTreeReadPool.h:257]: Can you just add name of the class before initializer?

```suggestion
        extension.all_callback(InitialAllRangesAnnouncement{
```

you have to jump couple of times to get to the code which tells you which class is constructed.

**Review by antonio2368 — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeReadPool.h [hunk 3]
```diff
@@ -194,8 +194,9 @@ class MergeTreeReadPoolParallelReplicas : public IMergeTreeReadPool
             actions_settings, reader_settings, per_part_params);
 
         extension.all_callback({
```
Comment (nikitamikhaylov) [inline src/Storages/MergeTree/MergeTreeReadPool.h:196]:
```suggestion
        extension.all_callback(InitialAllRangesAnnouncement{
```

**Comment by nikitamikhaylov** [comment]
A PR which fixes `test_disk_over_web_server` is already merged to https://github.com/ClickHouse/ClickHouse/pull/51494

**Pull request merged**